internal/lsp/format.go

```diff
@@ -0,0 +1,49 @@
+package lsp
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-policy-agent/opa/format"
+)
+
+func Format(path, contents string, opts format.Opts) (string, error) {
+	formatted, err := format.SourceWithOpts(filepath.Base(path), []byte(contents), opts)
+	if err != nil {
+		return "", fmt.Errorf("failed to format Rego source file: %w", err)
+	}
+
+	return string(formatted), nil
+}
+
+// ComputeEdits computes diff edits from 2 string inputs.
+func ComputeEdits(before, after string) []TextEdit {
```

Comment:
This was borrowed from regols, so we should give some credit (and a MIT header, I guess) here.

Reply:
Like this: c1a079681e395948f70f9123583d47a70a8321b2